igvc_navigation/src/path_follower/path_follower.cpp

```diff
@@ -157,17 +164,43 @@ void PathFollower::positionCallback(const nav_msgs::OdometryConstPtr& msg)
 
 nav_msgs::PathConstPtr PathFollower::getPatchedPath(const nav_msgs::PathConstPtr& msg) const
 {
-  nav_msgs::PathPtr new_path = boost::make_shared<nav_msgs::Path>(*msg);
-  size_t num_poses = new_path->poses.size();
-  for (int i = 0; i < num_poses; i++)
+  // Patch heading into each point
+  nav_msgs::Path new_path = *msg;
+  size_t num_poses = new_path.poses.size();
+  for (size_t i = 0; i < num_poses; i++)
   {
-    double delta_x = new_path->poses[i + 1].pose.position.x - new_path->poses[i].pose.position.x;
-    double delta_y = new_path->poses[i + 1].pose.position.y - new_path->poses[i].pose.position.y;
+    double delta_x = new_path.poses[i + 1].pose.position.x - new_path.poses[i].pose.position.x;
```

Comment:
I think that calculating the headings for the original path, then calculating a new path, `smoothed_path`, then calculating the new headings for that path is a bit wasteful.

Why not make `smoothed_path` your only path to begin with and avoid calculating headings for the original path.